t/t5516-fetch-push.sh

```diff
@@ -1712,4 +1712,23 @@ test_expect_success 'updateInstead with push-to-checkout hook' '
 	)
 '
 
+test_expect_success 'denyCurrentBranch and worktrees' '
+    git worktree add new-wt &&
```

Comment:
There is some funny indentation going on here; It is different from the other lines'.

Apart from that, the commit message starts with "t5516: denyCurrentBranch and unborn branch", which leaves almost everything about the patch unclear. What is the goal of this patch? Is it a bug fix? Does it describe the scenario better? What is t5516? 

Also, are you sure that you want to add this test case in _this_ commit? This is the regression test protecting the other bug fix from being broken again in the future, no? I.e. the "denyCurrentBranch should respect _all_ worktrees" one, yet this commit should be about the "demonstrate denyCurrentBranch/unborn branch bug ".